project/web/balancer.sls

```diff
@@ -170,6 +170,7 @@ install_letsencrypt:
   git.latest:
     - name: https://github.com/letsencrypt/letsencrypt/
     - target: {{ letsencrypt_dir }}
+    - force_checkout: True
```

Comment:
The Salt docs say this should have done it:

    force_checkout : False
        When checking out the local branch, the state will fail if there are unwritten
        changes. Set this argument to True to discard unwritten changes when checking out.